src/uct/ib/base/ib_iface.c

```diff
@@ -228,27 +234,31 @@ size_t uct_ib_address_size(const union ibv_gid *gid, uint8_t is_global_addr,
     if (is_link_layer_eth) {
         return sizeof(uct_ib_address_t) +
                sizeof(union ibv_gid);  /* raw gid */
-    } else if ((gid->global.subnet_prefix == UCT_IB_LINK_LOCAL_PREFIX) &&
-               !is_global_addr) {
+    }
+
+    if (!is_global_addr) {
         return sizeof(uct_ib_address_t) +
                sizeof(uint16_t); /* lid */
-    } else if (((gid->global.subnet_prefix & UCT_IB_SITE_LOCAL_MASK) ==
-                UCT_IB_SITE_LOCAL_PREFIX) && !is_global_addr) {
+    }
+
+    if ((gid->global.subnet_prefix & UCT_IB_SITE_LOCAL_MASK) ==
+        UCT_IB_SITE_LOCAL_PREFIX) {
         return sizeof(uct_ib_address_t) +
                sizeof(uint16_t) + /* lid */
                sizeof(uint64_t) + /* if_id */
                sizeof(uint16_t);  /* subnet16 */
-    } else {
-        return sizeof(uct_ib_address_t) +
-               sizeof(uint16_t) + /* lid */
-               sizeof(uint64_t) + /* if_id */
-               sizeof(uint64_t);  /* subnet64 */
     }
+
+    return sizeof(uct_ib_address_t) +
+           sizeof(uint16_t) + /* lid */
+           sizeof(uint64_t) + /* if_id */
+           sizeof(uint64_t);  /* subnet64 */
```

Comment:
i think it would be better to keep "..else if" patter as before, because it's not err handling, it's different cases which are "equal" in importance

Reply:
ok